### PR TITLE
Fix Realm cache during migration

### PR DESCRIPTION
**Pull request by cmelchior**
This makes sure that Realms aren't cached if a RealmMigrationNeededException is thrown. It could result in references to wrong or closed Realms being returned on next call to Realm.getInstance(). Reported here: https://github.com/realm/realm-java/issues/765

@emanuelez @kneth @bmunkholm

### CODE REVIEW

**Comment by kneth** [comment]
:+1:

File: realm/src/main/java/io/realm/internal/Group.java [hunk 1]
```diff
@@ -215,12 +215,15 @@ public String getTableName(int index) {
      */
     public Table getTable(String name) {
         verifyGroupIsValid();
-        if (name == null || name.equals(""))
+        if (name == null || name.equals("")) {
```
Comment (emanuelez) [inline realm/src/main/java/io/realm/internal/Group.java]: name.isEmpty() is better

**Comment by emanuelez** [comment]
Minor comment, otherwise :+1:

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,7 @@
+0.78
+ * Fixed bug causing Realms to be cached during a RealmMigration, resulting in invalid realms being returned.
```
Comment (bmunkholm) [inline changelog.txt]: The description is better in the description of this PR :-)
e.g. "realms being returned" - from where?

**Comment by bmunkholm** [comment]
Otherwise :+1:

**Pull request merged**